presto-native-execution/presto_cpp/main/SystemConnector.cpp

```diff
@@ -0,0 +1,392 @@
+        break;
+      }
+
+      case TaskColumnEnum::kRunningSplits: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.runningDrivers);
+        break;
+      }
+
+      case TaskColumnEnum::kCompletedSplits: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.completedDrivers);
+        break;
+      }
+
+      case TaskColumnEnum::kSplitScheduledTimeMs: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(toMillis(taskInfo.stats.totalScheduledTimeInNanos));
+        break;
+      }
+
+      case TaskColumnEnum::kSplitCpuTimeMs: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(toMillis(taskInfo.stats.totalCpuTimeInNanos));
+        break;
+      }
+
+      case TaskColumnEnum::kSplitBlockedTimeMs: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(toMillis(taskInfo.stats.totalBlockedTimeInNanos));
+        break;
+      }
+
+      case TaskColumnEnum::kRawInputBytes: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.rawInputDataSizeInBytes);
+        break;
+      }
+
+      case TaskColumnEnum::kRawInputRows: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.rawInputPositions);
+        break;
+      }
+
+      case TaskColumnEnum::kProcessedInputBytes: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.processedInputDataSizeInBytes);
+        break;
+      }
+
+      case TaskColumnEnum::kProcessedInputRows: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.processedInputPositions);
+        break;
+      }
+
+      case TaskColumnEnum::kOutputBytes: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.outputDataSizeInBytes);
+        break;
+      }
+
+      case TaskColumnEnum::kOutputRows: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.outputPositions);
+        break;
+      }
+
+      case TaskColumnEnum::kPhysicalWrittenBytes: {
+        auto flat = result->childAt(i)->as<FlatVector<int64_t>>();
+        SET_TASK_COLUMN(taskInfo.stats.physicalWrittenDataSizeInBytes);
+        break;
+      }
+
+      case TaskColumnEnum::kCreated: {
+        auto flat = result->childAt(i)->as<FlatVector<Timestamp>>();
+        SET_TASK_COLUMN(velox::Timestamp::fromMillis(task->createTimeMs));
+        break;
+      }
+
+      case TaskColumnEnum::kStart: {
+        auto flat = result->childAt(i)->as<FlatVector<Timestamp>>();
+        SET_TASK_COLUMN(
+            velox::Timestamp::fromMillis(task->firstSplitStartTimeMs));
```

Comment:
Why is this not `taskInfo.stats.firstStartTime`?
Same above `taskInfo.stats.createTime`?

Reply:
taskInfo.stats.firstStartTime and createTime are serialized std::strings and we would need to re-parse them here. So instead we went with maintaining the original u_int64t ms values.